tests/MdTablesSpec.hs

```diff
@@ -0,0 +1,238 @@
+      it "returns nothing for table without hr" $
+        getTable tableWOBreakMD `shouldBe` Nothing
+      it "returns table without column names" $
+        getTable tableWOColumnNamesMD `shouldBe` Just tableWOColumnNames
+    describe "rendering" $ do
+      it "renders simple table with column names" $ do
+        renderedT <- render simpleTable
+        renderedT `shouldBe` T.toLazy simpleTableHtml
+      it "renders simple table without column names" $ do
+        renderedT <- render tableWOColumnNames
+        renderedT `shouldBe` T.toLazy tableWOColumnNamesHtml
+      it "renders full table example" $ do
+        renderedT <- render fullTable
+        renderedT `shouldBe` T.toLazy fullTableHtml
+
+render = liftIO . renderTextT . renderTable
+
+-----------------------
+------ Text Md --------
+-----------------------
+
+tableKeyword :: Text
+tableKeyword = "+ %TABLE Table"
+
+columnNamesList :: [Text]
+columnNamesList =
+  [ "+ - Column 1"
+  , "  - Column 2"
+  , "  - Column 3"
+  ]
+
+columnNamesSimple :: Text
+columnNamesSimple = "+ - Column 1|Column 2|Column 3"
+
+breakText :: Text
+breakText = "+ ----------------"
+
+rowOfList :: [Text]
+rowOfList =
+  [ "+ - Foo"
+  , "  - Bar"
+  , "  - Baz"
+  , ""
+  ]
+
+rowWithSeparator :: Text
+rowWithSeparator = "+ - Another foo | Another bar | Another baz"
+
+rowSeveralNodes :: Text
+rowSeveralNodes = "+ - *foo* | **bar** | baz"
+
+rowOfListSeveralNodes :: [Text]
+rowOfListSeveralNodes =
+  [ "+ - ```"
+  , "    Code foo"
+  , "    ```"
+  , "  - Simple bar"
+  , "  - `inline code` baz"
+  ]
+
+---------------------------
+------- MD elements -------
+---------------------------
+
+columnNames :: [[MD.Node]]
+columnNames = [ [ MD.Node Nothing (MD.TEXT "Column 1") [] ]
+              , [ MD.Node Nothing (MD.TEXT "Column 2") [] ]
+              , [ MD.Node Nothing (MD.TEXT "Column 3") [] ]
+              ]
+
+rowOfListMD :: [[MD.Node]]
+rowOfListMD = [ [ MD.Node Nothing (MD.TEXT "Foo") [] ]
+              , [ MD.Node Nothing (MD.TEXT "Bar") [] ]
+              , [ MD.Node Nothing (MD.TEXT "Baz") [] ]
+              ]
+
+rowSeveralNodesMD :: [[MD.Node]]
+rowSeveralNodesMD = [ [ MD.Node Nothing MD.EMPH [ MD.Node Nothing (MD.TEXT "foo") [] ]
+                      , MD.Node Nothing (MD.TEXT " ") []
+                      ]
+                    , [ MD.Node Nothing (MD.TEXT " ") []
+                      , MD.Node Nothing MD.STRONG [ MD.Node Nothing (MD.TEXT "bar") [] ]
+                      , MD.Node Nothing (MD.TEXT " ") []
+                      ]
+                    , [ MD.Node Nothing (MD.TEXT " baz") [] ]
+                    ]
+
+rowWithSeparatorMD :: [[MD.Node]]
+rowWithSeparatorMD = [ [ MD.Node Nothing (MD.TEXT "Another foo ") [] ]
+                     , [ MD.Node Nothing (MD.TEXT " Another bar ") [] ]
+                     , [ MD.Node Nothing (MD.TEXT " Another baz") [] ]
+                     ]
+
+rowOfListSeveralNodesMD :: Int -> [[MD.Node]]
+rowOfListSeveralNodesMD n =
```

Comment:
It's specialty of the test example. Manipulations with lines in Markdown makes positions changed, and `==` in `shoulBe` won't work without certain position specifying.